src/applications/applications.service.ts

```diff
@@ -10,21 +10,34 @@ import {
   ApplicationDto,
   UpdateApplicationDto,
 } from 'applications/dto';
-import { applications } from 'data-services/data-services-mocked/data/mock.applications';
-import { RequestService } from 'requests/requests.service';
+import { IApplicationRepository } from 'data-services/interfaces/iapplication-repository';
+import { IDataServices } from 'data-services/interfaces/idata-services';
+import { IRequestRepository } from 'data-services/interfaces/irequest-repository';
 
 @Injectable()
 export class ApplicationService {
-  constructor(private requestService: RequestService) {}
+  constructor(dataServices: IDataServices) {
+    this.applications = dataServices.applications;
+    this.requests = dataServices.requests;
+  }
+
+  private applications: IApplicationRepository;
+
+  private requests: IRequestRepository;
 
   private notFoundMsg = 'Application not found';
 
-  private getAll() {
-    return applications;
+  private async getAll() {
+    return this.applications.getAll();
+  }
+
+  private async isRequestActual(id: string) {
+    const request = await this.requests.getById(id);
```

Comment:
You should use getById from request service, not repository, because you can have null or undefined which are not handled by repository with exceptions